prody/dynamics/plotting.py

```diff
@@ -2381,3 +2355,43 @@ def showTree_networkx(tree, node_size=20, node_color='red', node_shape='o',
         showFigure()
 
     return mpl.gca()
+
+
+def checkColors(colors, num, labels, allowNumbers=False):
+    """Check colors and process them if needed"""
+
+    from matplotlib.colors import is_color_like
+
+    colors_dict = {}
+
+    if isinstance(colors, np.ndarray):
+        colors = tuple(colors)
+
+    if is_color_like(colors) or colors is None:
+        colors = [colors] * num
+    elif isListLike(colors):
+        colors = list(colors)
+
+    if isinstance(colors, dict):
+        if labels is None:
+            raise TypeError('color must be a string or a list unless labels are provided')
+        colors_dict = colors
+        colors = [colors_dict[label] for label in labels]
+
+    if isinstance(colors, list):
+        if len(colors) != num and not is_color_like(colors):
+            raise ValueError('colors should have the length of the set to be colored or satisfy matplotlib color rules')
+
+        if np.any([not is_color_like(color) for color in colors]):
```

Comment:
Please consider using a for-loop instead of `np.any` and list comprehension for better readability. The logic seems to be a bit convoluted for what it does. Especially since there's another list comprehension in line 2388.

Reply:
wouldn't a for loop be much slower?

What's not clear about it?

Reply:
List comprehension is slightly faster than a for-loop for **building a list**. List comprehension isn't that much different than a for-loop in terms of performance. Here, since we don't need to build a list, the code below is faster for two reasons:
1. no need to build a list for `np.any`, which requires unnecessary memory and time for the numpy array.
2. you get an early exit when one of the `colors` is wrong, so that you don't need to check the rest of them.
3. avoided a for-loop since in line 2381.

```
for color in colors:
    if not is_color_like(color):
        if not allowNumbers:
            raise ValueError('each element of colors should satisfy matplotlib color rules')
        elif not isinstance(color, Number):
            raise ValueError('each element of colors should be a number or satisfy matplotlib color rules')
```

Regarding readability, most of time list comprehension can increase readability, but sometimes it can overcomplicate things, especially if it is used in an `if` statement.

Reply:
ok, that makes sense. Done